code/modules/client/preferences.dm

```diff
@@ -167,6 +167,7 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 		tainted_character_profiles = FALSE
 
 	data["character_preferences"] = compile_character_preferences(user)
+	data["character_preferences"]["all_preferences"] = compile_character_preferences(user, FALSE)
```

Comment:
This is extremely wasteful and I don't completely understand why we're doing it like this.

Reply:
fair point, theres a much much more elegant way to do it, i dont remember why i did it this way to begin with

Reply:
fixed